### PR TITLE
 implement "--workdir"

### PR DESCRIPTION
**Pull request by sh4wn**
- implement "--workdir"
- fix test failure on Docker Desktop for macOS

### CODE REVIEW

File: runlike/inspector.py [hunk 1]
```diff
@@ -90,6 +95,14 @@ def parse_ports(self):
                         '--expose=%s' %
                         container_port_and_protocol)
 
+        exposed_ports = self.get_fact("Config.ExposedPorts")
+        if exposed_ports:
```
Comment (pox) [inline runlike/inspector.py:99]:
```suggestion
        if exposed_ports and not ports:
```

File: runlike/inspector.py [hunk 2]
```diff
@@ -90,6 +95,14 @@ def parse_ports(self):
...
+            for container_port_and_protocol, options in exposed_ports.items():
+                if not ports or container_port_and_protocol not in ports.keys():
```
Comment (pox) [inline runlike/inspector.py:101]:
```suggestion
                if container_port_and_protocol not in ports.keys():
```

File: runlike/inspector.py [hunk 3]
```diff
@@ -155,6 +168,10 @@ def parse_extra_hosts(self):
         hosts = self.get_fact("HostConfig.ExtraHosts") or []
         self.options += ['--add-host %s' % host for host in hosts]
 
+    def parse_workdir(self):
+        workdir = self.get_fact("Config.WorkingDir")
+        self.options.append("--workdir=%s" % workdir)
```
Comment (pox) [inline runlike/inspector.py:173]: What if `workdir` is `None`?

File: test_runlike.py [hunk 4]
```diff
@@ -152,3 +152,6 @@ def test_cap_add(self):
     def test_devices(self):
         self.expect_substr("--device /dev/null:/dev/null:r")
         self.expect_substr("--device /dev/null:/dev/null", 2)
+
+    def test_workdir(self):
+        self.expect_substr("--workdir=/workdir")
```
Comment (pox) [inline test_runlike.py:157]: I think it's important to add a negative test here as well (expect not to find `--workdir` in the printout if it hasn't been set).

**Review by pox — COMMENTED** [review]
Thank you for the contribution, and my apologies for taking so long to review them.

**Comment by lavie** [comment]
When trying to run `make test` for this PR I get a container that hangs (`fixtures1`) on Ubuntu. The container cannot be deleted or killed once it's created.
Very odd, and probably a bug in docker. But in any case, I can't very well merge this under these circumstances.
@sh4wn could you please try running the tests yourself and see if the problem is recreated]?

**Comment by lavie** [comment]
@sh4wn please refrain from force-pushing additional changes to this branch because it makes it hard to follow this conversation and see what was changed since the last review. Once this PR is accepted we can always do a cleanup rebase and squash stuff to make it all pretty.

File: runlike/inspector.py (hunk 1 above)
Comment (sh4wn) [reply runlike/inspector.py:99]: I believe "NetworkSettings.Ports" and "Config.ExposedPorts" may both exist.

**Comment by sh4wn** [comment]
> @sh4wn please refrain from force-pushing additional changes to this branch because it makes it hard to follow this conversation and see what was changed since the last review. Once this PR is accepted we can always do a cleanup rebase and squash stuff to make it all pretty.

**Comment by sh4wn** [comment]
> > @sh4wn please refrain from force-pushing additional changes to this branch because it makes it hard to follow this conversation and see what was changed since the last review. Once this PR is accepted we can always do a cleanup rebase and squash stuff to make it all pretty.

Sorry,it's my first time to create a PR.

**Comment by lavie** [comment]
@sh4wn no problem. I appreciate the contribution.

Let me know if you manage to recreate the testing problem. On my Ubuntu I cannot run tests for this branch - and only for this branch. It's very peculiar.

**Comment by sh4wn** [comment]
@lavie The "--workdir" option changed the working directory,So it can not find the 'sleep.sh' file.

File: runlike/inspector.py (hunk 2 above)
Comment (sh4wn) [reply runlike/inspector.py:101]: The binding ports and exposed ports may both appear in "Config.ExposedPorts", but only the  binding ports will appear in "NetworkSettings.Ports". So i think the code is correct.

**Comment by sh4wn** [comment]
@lavie The docker image assaflavie/runlike:0.7.0 has the pip/runlike 0.6.2 installed. could you update the docker image?

**Comment by lavie** [comment]
@sh4wn fixed! Thanks for noticing.
